---
title: St. Jude Cloud Documentation
---

Welcome to the St. Jude Cloud documentation! Here, you will find authoritative guides for accessing St. Jude Cloud data, creating and running analysis workflows on the cloud, and exploring curated data from numerous published studies by St. Jude and our collaborating institutions.To learn more about our ecosystem, read this [guide](./ecosystem).

Sign up [here](https://hospital.stjude.org/apps/forms/fb/st-jude-cloud-subscribe/) to receive email notifications when we add new datasets, analysis pipelines, or other exciting features.

## Features

You can apply many different capabilities of St. Jude Cloud to your research, such as:

* Explore the [raw genomics data](https://platform.stjude.cloud/data/diseases) we currently offer. You can browse by diagnosis, publication, or curated dataset while applying a number of different filters. For more information, see our [data request guide](./genomics-platform/requesting-data/making-a-data-request).
* Run your tools on *our* data by [requesting data](./genomics-platform/requesting-data/making-a-data-request) and packaging your tools in a secure cloud environment. See this [guide](./genomics-platform/analyzing-data/creating-a-cloud-app) for an example.
* Run our in house analysis workflows on *your* data by [moving your data](./genomics-platform/managing-data/upload-local) to the cloud and selecting a [workflow](https://platform.stjude.cloud/workflows) to run. See this [guide](./genomics-platform/analyzing-data/running-sj-workflows) for an example. 
* Explore St. Jude datasets through interactive visualizations that we have packaged for the community. For example, visit [PeCan](https://pecan.stjude.cloud) to visually investigate pediatric cancer mutation data.
* Create manuscript quality figures with *your* data to use in publications or to host on your website with [ProteinPaint](https://proteinpaint.stjude.org/) or [GenomePaint](https://genomepaint.stjude.cloud/). See the [ProteinPaint documentation](./visualization-community/proteinpaint) and [GenomePaint documentation](./visualization-community/genomepaint) for help.

!!!note
Please note that while it is free to receive and store our data in St. Jude Cloud, there are [compute and storage fees associated with working in the cloud, as well as egress fees for downloading data](./faq/#will-i-be-charged-for-using-st-jude-cloud). 
!!!

## Apps

St. Jude Cloud provides a number of applications which you can use for various purposes. Click on the links below to checkout one or more of our applications.

* [Genomics Platform](https://platform.stjude.cloud): browse our publicly available genomics datasets, make a data request, and run analyses in a secure cloud environment. View the documentation for this app [here](./genomics-platform). 

* [Pediatric Cancer Portal (PeCan)](https://pecan.stjude.cloud): interactively explore mutational recurrence and pathogenicity assessment of variants in pediatric cancer using a wide variety of St. Jude + publicly available data. View the documentation for this app [here](./pecan).

* [Visualization Community](https://viz.stjude.cloud/): explore, create, and share interactive visualizations using tools such as ProteinPaint and GenomePaint. View the documentation for this app [here](./visualization-community).

## Studies

The following projects currently distribute data through St. Jude Cloud. Click [here](./genomics-platform/requesting-data/about-our-data/#data-access-units) for a brief description of each project listed below. Or click on a project in the bulleted list below to go straight to the corresponding Study page.

* [Pediatric Cancer Genome Project (PCGP)](https://stjude.cloud/studies/pediatric-cancer-genome-project)
* [St. Jude Lifetime (SJLIFE)](https://sjlife.stjude.org/)
* [Clinical Genomics (Clinical Pilot and G4K)](https://stjude.cloud/studies/clinical-genomics)
* [Sickle Cell Genome Project (SGP)](https://sickle-cell.stjude.cloud)
* [Childhood Cancer Survivor Study (CCSS)](https://ccss.stjude.org/)

## Analysis Workflows

St. Jude shares a number of end-to-end analysis workflows as part of the Genomics Platform application. Click on the links below to learn more about the workflow.

<!-- * [NeoepitopePred](https://platform.stjude.cloud/workflows/neoepitopepred) -->
* [ChIP-Seq Peak Calling](https://platform.stjude.cloud/workflows/chip-seq)
* [Rapid RNA-Seq Fusion Detection](https://platform.stjude.cloud/workflows/rapid_rna-seq)
* [WARDEN Differential Expression Analysis](https://platform.stjude.cloud/workflows/warden)
* [Mutational Signatures](https://platform.stjude.cloud/workflows/mutational_signatures)
* [SequencErr](https://platform.stjude.cloud/workflows/sequencerr)
* [MethylationToActivity (M2A)](https://platform.stjude.cloud/workflows/methylation-to-activity)
* [RNA-Seq Expression Classification](https://platform.stjude.cloud/workflows/rnaseq-expression-classification)
* [cis-X](https://platform.stjude.cloud/workflows/cis-x)

## Contact Us

Any questions, comments, or concerns can be directed to our ["Contact Us"](https://stjude.cloud/contact) form or you can email us directly at support@stjude.cloud.
